torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1341,13 +1385,21 @@ void ProcessGroupNCCL::heartbeatMonitor() {
   // Case two: desync might be slow or get stuck. Or we get stuck in
   // destructors, we will sleep for some time before calling std::abort() to
   // kill the whole process.
-  if ((terminateProcessGroup_.load() || collectiveDebugInfoMode_.load()) &&
+  if ((terminateProcessGroup_.load() || collectiveDebugInfoMode_.load() ||
+       !debugInfoStoreThread.joinable()) &&
```

Comment:
is it possible joinable is false here? under which conditions?

Reply:
When the thread has not finished or not initialized, `joinable` return False.

Reply:
Hmm this condition should be `debugInfoStoreThread.joinable()` I got confused `joinable` with the end of thread execution previously..